runtime/ts/storage/storage-provider-factory.ts

```diff
@@ -8,12 +8,12 @@
 
 import {InMemoryStorage} from './in-memory-storage';
 import {FirebaseStorage} from './firebase-storage';
+import {Id} from '../id';
 
 export class StorageProviderFactory {
   _storageInstances: {[index: string]: InMemoryStorage | FirebaseStorage};
-  _arcId: string;
-  constructor(arcId) {
-    this._arcId = arcId;
+
+  constructor(private readonly arcId:Id) {
```

Comment:
Is there a Google rule about this? Naively, I quite like having all the data members declared at the top of the class.

Reply:
Hmm, or can you not use readonly if you declare it up top?

Reply:
oh this is similar to C++ constructors is it? Hmm. That makes it a trickier call.

Is there a lint rule that requires the constructor to be the first function, and all non-constructor class members to be declared immediately above? If so then using the constructor to declare members like this is probably better. I suppose we can probably just require this even if we can't lint it yet.

Otherwise, I worry a bit about the class declaration becoming meaningless in practice as you can't see what's in a class at a glance.